src/EnergyPlus/SwimmingPool.cc

```diff
@@ -953,6 +944,26 @@ void SwimmingPoolData::calculate(EnergyPlusData &state)
     state.dataHeatBalFanSys->SumLatentPool(ZoneNum) += EvapRate * Psychrometrics::PsyHfgAirFnWTdb(thisZoneHB.airHumRat, thisZoneHB.MAT);
 }
 
+void SwimmingPoolData::calcMassFlowRate(EnergyPlusData &state, Real64 &massFlowRate, Real64 TH22, Real64 TLoopInletTemp)
```

Comment:
@rraustad Here are answers to your questions:

1. TH22 comes from the temperature histories array.  One of the twos is for the inside surface and the other is for the last temperature history (not the current time).  So, yes, it is the pool water temperature and I guess it could have been better stated that way.  It's also TH22 in the routine that calls it so I was trying to keep some continuity there to make it less confusing, but I can see that I perhaps wasn't successful since the meaning of TH22 is not obvious.
2. That could be done.  I guess I was calculating and fixing it at a scalar so that I wasn't using the same equation in both places.  My thought was: I calculated this by hand so that it would be correct--does the code correctly calculate this using the subroutine?
3. Actually, it is possible for this to calculate a negative number, but this is why there is the else if starting in line 957.  These lines of code prevent returning a negative flow and also make sure that if the loop temperature is lower than the setpoint that it still sends flow in the case where heating could actually be taking place.  Previously, it just set the flow to zero which is what led to this defect.  Now, when the flow gets calculated as negative or zero, it checks to see if the loop temperature would still provide heating (that is, is higher than the pool temperature) even though the loop temperature is lower than the setpoint temperature.  Before, the flow was set to zero and a low capacity heater would just quit heating.  Now, the low capacity heater still runs and can eventually get back to the setpoint.

@Myoldmopar Based on comments by @rraustad, do you want me to do any additional commits?

Reply:
@RKStrand my comments were more for closure on this branch. I don't really "need" to see these changes, they were just  food for thought.

Reply:
Perfect on all accounts.  They are valid comments, but I agree we don't need more commits here.  Thanks to you both, I think this is ready.

Reply:
Ready - "was ready when I merged it"